NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -251,8 +510,567 @@ def make_sim_shower(corsika, observer=None, detector=None, station_id=None):
         sim_shower.set_parameter(shp.electromagnetic_energy, corsika["highlevel"].attrs["Eem"] * units.eV)
     except:
         global warning_printed_coreas_py
-        if(not warning_printed_coreas_py):
-            logger.warning("No high-level quantities in HDF5 file, not setting EM energy, this warning will be only printed once")
+        if (not warning_printed_coreas_py):
+            logger.warning(
+                "No high-level quantities in HDF5 file, not setting EM energy, this warning will be only printed once")
             warning_printed_coreas_py = True
 
     return sim_shower
+
+
+def create_sim_shower(evt, detector=None, station_id=None):
+    """
+    creates an NuRadioReco sim shower from an event object created with e.g. read_CORSIKA7(),
+    the core positions are set such that the detector station is on top of
+    each coreas observer position
+
+    Parameters
+    ----------
+    evt : Event object
+        event object containing the CoREAS output, e.g. created with read_CORSIKA7()
+    detector : detector object
+    station_id : int
+        station id of the station relative to which the shower core is given
+
+    Returns
+    -------
+    sim_shower: sim shower
+        simulated shower object
+    """
+    sim_shower = copy.copy(evt.get_first_sim_shower())
+
+    efields = evt.get_station(0).get_sim_station().get_electric_fields()
+
+    # We can only set the shower core relative to the station if we know its position
+    if efields is not None and detector is not None and station_id is not None:
+        efield_pos = []
+        for efield in efields:
+            efield_pos.append(efield.get_position())
+        efield_pos = np.array(efield_pos)
```

Comment:
This is from before we had the interpolation. The idea was, that one uses a shower n_obs times to calculate the effective area. So to avoid interpolation we place the core in a way, that a certain observer is on top of a single detector station. I agree, that this is confusing but it serves its purpose. For this, also the calculate_simulation_weights() function was used. I will follow up in another comment.